firstvoices/backend/models/media.py

```diff
@@ -208,51 +245,6 @@ def generate_resized_images(self):
                 ),
             )
 
-    def save(self, **kwargs):
-        # Boolean which returns True if the image is new, otherwise false.
-        is_new = self._state.adding is True
-        # Boolean which returns True if the image file has been updated, otherwise false (returns true for a new image).
-        content_updated = hasattr(self.content, "file") and isinstance(
-            self.content.file, UploadedFile
-        )
-
-        # If the main image file was updated then delete the old image files from AWS
-        if content_updated and not is_new:
-            try:
-                old_image = Image.objects.get(pk=self.pk)
-                old_image.content.delete(save=False)
-                self.thumbnail.delete(save=False)
-                self.small.delete(save=False)
-                self.medium.delete(save=False)
-            except Exception as e:
-                # this will only happen for connection or permission errors, so it's a warning
-                self.logger.warn(
-                    f"Failed to delete file from S3 when deleting [{str(self)}]. Error: {e} "
-                )
-
-        # If the image is new or the file has been updated then generate new resized images
-        if is_new or content_updated:
-            self.generate_resized_images()
-
-        super().save()
-
-    def delete(self, using=None, keep_parents=False):
-        """
-        Deletes the associated media files when the instance is deleted, to prevent orphans.
-        """
-        result = super().delete(using, keep_parents)
-        try:
-            self.thumbnail.delete(save=False)
-            self.small.delete(save=False)
-            self.medium.delete(save=False)
-        except Exception as e:
-            # this will only happen for connection or permission errors, so it's a warning
-            self.logger.warn(
-                f"Failed to delete file from S3 when deleting [{str(self)}]. Error: {e} "
-            )
-
-        return result
```

Comment:
Moved these to the `MediaBase` class as they are needed for both the `Image` and `Video` models.